### ISSUE TITLE
Change License to Apache-2.0

### ISSUE THREAD

**Issue by ahoarau** [orocos/orocos_kinematics_dynamics#405]
Hi there !

In the context of a private company with proprietary code, having an LGPLv2 can be problematic.
The biggest concern is that it prevents from statically link against the library easily.

I propose to change the license to the more permissive **Apache-2.0**, as you did for all ros2 related stuff 
* https://github.com/orocos/rtt_ros2_integration/blob/master/LICENSE
* https://github.com/orocos/rtt_ros2_common_interfaces/blob/master/LICENSE.

They did the license change in PlotJuggler: https://github.com/facontidavide/PlotJuggler/issues/690
Basically we create an issue and wait for everyone to accept the change.

@MatthijsBurgh What do you think ?

PS: Other licenses might just work as well (MIT)

**Comment by traversaro** [comment]
Indeed, if that is ok for KDL mantainers it would be great!

**Comment by MatthijsBurgh** [comment]
I leave this one to decide to @smits @meyerj.

Also this requires the header of all files to be changed.

**Comment by smits** [comment]
@MatthijsBurgh this is ok from my side, you might want to check if there are files that have copyrights not owned by me.

**Comment by MatthijsBurgh** [comment]
Yes these files are definitely there.
I am sorry, but I don't feel the responsibility to get consent from all the copyright holders. I think that is up to @ahoarau and @traversaro to do it

**Comment by ahoarau** [comment]
ok @MatthijsBurgh I'm closing this to re-open a new one. I'm gonna notify all contributors, not only headers names. I believe this is necessary for legal reasons. 
Also, the MIT license seems more appropriate. That "notice code changes" is cumbersome.
